src/core/server/server.ts

```diff
@@ -215,7 +220,7 @@ export class Server {
       coreId,
       'core',
       async (context, req, res): Promise<RequestHandlerContext['core']> => {
-        const savedObjectsClient = coreSetup.savedObjects.getScopedClient(req);
+        const savedObjectsClient = this.coreStart!.savedObjects.getScopedClient(req);
         const uiSettingsClient = coreSetup.uiSettings.asScopedToClient(savedObjectsClient);
```

Comment:
now uiSettings should remove `asScopedToClient` from setup contract for extra safety. I can do it in a separate PR.

Reply:
Created https://github.com/elastic/kibana/issues/55396